File: kotlin/src/main/kotlin/io/github/devstarsj/kotlinbackend/UserDto.kt
package io.github.devstarsj.kotlinbackend

import java.time.LocalDateTime

class UserDto {
    val id: Long?
    val username: String?
    val password: String?
    val items: List<ItemDto>
    val createdAt: LocalDateTime?
    val updatedAt: LocalDateTime?

    constructor(user: User?) {
        id = user?.id
        username = user?.username
        password = user?.password
        createdAt = user?.createdAt
        updatedAt = user?.updatedAt
        items = user?.items?.map { ItemDto(it) } ?: arrayListOf()
    }
}

File: nest.js/src/dtos/userDto.ts
import { ObjectType, Field } from '@nestjs/graphql'
import { ItemDto } from './itemDto'

@ObjectType()
export class UserDto {

  @Field()
  readonly id?: number

  @Field()
  username!: string

  @Field(type => [ItemDto], { nullable: true })
  items?: ItemDto[]

  @Field()
  createdAt: Date

  @Field()
  updatedAt: Date
}
